drivers/usb/usbccgp/function.c

```diff
@@ -654,24 +654,66 @@ USBCCGP_LegacyEnum(
             return STATUS_UNSUCCESSFUL;
         }
 
-        //
-        // init function descriptor
-        //
-        FDODeviceExtension->FunctionDescriptor[Index].FunctionNumber = Index;
-        FDODeviceExtension->FunctionDescriptor[Index].NumberOfInterfaces = 1;
-        FDODeviceExtension->FunctionDescriptor[Index].InterfaceDescriptorList = AllocateItem(NonPagedPool, sizeof(PUSB_INTERFACE_DESCRIPTOR) * 1);
-        if (!FDODeviceExtension->FunctionDescriptor[Index].InterfaceDescriptorList)
+        SubIndex = 0;
+        if (InterfaceDescriptor->bInterfaceClass == USB_DEVICE_CLASS_AUDIO)
         {
+            // AUDIO CLASS lets group all audio interfaces together
+            //
+            // init function descriptor
+            //
+            FDODeviceExtension->FunctionDescriptor[Index].FunctionNumber = Index;
+            FDODeviceExtension->FunctionDescriptor[Index].NumberOfInterfaces = 1;
+            FDODeviceExtension->FunctionDescriptor[Index].InterfaceDescriptorList = AllocateItem(NonPagedPool, sizeof(PUSB_INTERFACE_DESCRIPTOR) * FDODeviceExtension->ConfigurationDescriptor->bNumInterfaces);
+            if (!FDODeviceExtension->FunctionDescriptor[Index].InterfaceDescriptorList)
+            {
+                //
+                // no memory
+                //
+                FreeItem(FDODeviceExtension->FunctionDescriptor);
```

Comment:
Shouldn't previous `FDODeviceExtension->FunctionDescriptor[...].InterfaceDescriptorList`s be freed too, first?